### PR TITLE
CTCP-4543: CREATE PAGE FOR COMMODITY CODE

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/controllers/houseConsignment/CommodityCodeController.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          case Some(value) => formProvider(houseConsignmentIndex, itemIndex).fill(value)
+        }
+
+        Ok(view(preparedForm, request.userAnswers.mrn, arrivalId, houseConsignmentIndex, itemIndex, isXI, mode))
+    }
+
+  def onSubmit(arrivalId: ArrivalId, houseConsignmentIndex: Index, itemIndex: Index, mode: Mode): Action[AnyContent] =
+    actions.getStatus(arrivalId).async {
+      implicit request =>
+        val isXI: Boolean = true
```
Comment (jackheslop96) [inline app/controllers/houseConsignment/CommodityCodeController.scala:59]: This doesn't look right?

Perhaps the isXI method could be extracted to something like
```
  private def isXI(implicit request: DataRequest[_]): Boolean =
    request.userAnswers.ie043Data.CustomsOfficeOfDestinationActual.referenceNumber.startsWith("XI")
```

File: app/controllers/houseConsignment/CommodityCodeController.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.houseConsignment
+
+import controllers.actions._
+import forms.CommodityCodeFormProvider
+import models.P5.{CustomsOfficeOfDestinationActual, MessageData}
```
Comment (jackheslop96) [inline app/controllers/houseConsignment/CommodityCodeController.scala:21]: Is `MessageData` being used?

File: app/controllers/houseConsignment/CommodityCodeController.scala (hunk 1 above)
Comment (kenneth-rayner) [reply app/controllers/houseConsignment/CommodityCodeController.scala]: Done

File: test/base/SpecBase.scala [hunk 3]
```diff
@@ -66,6 +66,7 @@ trait SpecBase
   val houseConsignmentIndex: Index  = Index(0)
   val itemIndex: Index              = Index(0)
   def emptyUserAnswers: UserAnswers = UserAnswers(arrivalId, mrn, eoriNumber, basicIe043, Json.obj(), Instant.now())
+  val isXI: Boolean                 = true
```
Comment (jackheslop96) [inline test/base/SpecBase.scala:69]: I would prefer to see this removed and dedicated unit tests for when this is true and when this is false

File: test/views/houseConsignment/CommodityCodeViewSpec.scala [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  behave like pageWithSubmitButton("Continue")
+
+  if (isXI) {
+
+    behave like pageWithContent(
+      "p",
+      "The combination of your commodity code and combined nomenclature code must be a valid code in TARIC. This is the European Union’s database for classifying goods and determining the amount of duties required."
+    )
+  }
```
Comment (jackheslop96) [inline test/views/houseConsignment/CommodityCodeViewSpec.scala:57]: We can have tests for when this is true and tests for when this is false. See example: https://github.com/hmrc/ctc-departure-transport-details-frontend/blob/744069d4bd6864d462a0cbc590009f2bd9ce9339/test/views/authorisationsAndLimit/authorisations/index/AuthorisationReferenceNumberViewSpec.scala#L65-L81

**Review by jackheslop96 — DISMISSED** [review]

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**